pkg/gadget-context/gadget-context.go

```diff
@@ -414,6 +401,49 @@ func (c *GadgetContext) LoadGadgetInfo(info *api.GadgetInfo, paramValues api.Par
 	return nil
 }
 
+func (c *GadgetContext) LoadGadgetInfoAndRun(info *api.GadgetInfo, paramValues api.ParamValues, remoteDone chan any) error {
+	// TODO: Is this needed here?
+	// This is already done in LoadGadgetInfo but we need to check it ourselves too if
+	// LoadGadgetInfoAndRun is called directly multiple times
+	//
+	// The naming suggests that this is only to check if the info was alreadty loaded
+	// but it is also used to check if the gadget was already started?
+	c.lock.Lock()
+	if c.loaded {
+		// TODO: verify that info matches what we previously loaded
+		c.lock.Unlock()
+		return nil
+	}
+	c.lock.Unlock()
+
+	err := c.LoadGadgetInfo(info, paramValues, nil)
+	if err != nil {
+		return err
+	}
+
+	if err := c.start(c.localOperators); err != nil {
+		return fmt.Errorf("starting local operators: %w", err)
+	}
+
+	c.Logger().Debugf("running...")
+
+	go func() {
+		// TODO: Client shouldn't need to wait for the timeout. It should be
+		// managed only on the server side.
+		WaitForTimeoutOrDone(c)
+
+		select {
+		case <-remoteDone:
+		// TODO Do we need a timeout here? I think not, since a second Ctrl+C does an exit
+		case <-time.After(3 * time.Second):
```

Comment:
The issue is that the client will remain stuck is the server never ends, however I think it's fine and I'd prefer not to introduce another arbitrary-defined timeout in our code.

Reply:
Thats what I thought. Also as the comment indicates, even if we have no timeout here, a second `Ctrl +C` should terminate the process anyway

Reply:
I changed by mind. I think we need to introduce this timeout to be totally sure this logic ends even if the server hangs. When running `kubectl-gadget` or `gadgetctl` pressing Ctrl + C twice does the trick, but it won't work if those binaries are executed by another binary without the logic to handle that case or if this code is used directly from Go.

cc @alban @flyth thoughts on this?

Reply:
No extra timeout needed, since we already have a default wait time for the server side to close in 
https://github.com/inspektor-gadget/inspektor-gadget/blob/e0f2eb7fa91346bb4a1e0bfc93e73f1da3143d9a/pkg/runtime/grpc/oci.go#L292